assimilation_code/modules/assimilation/adaptive_inflate_mod.f90

```diff
@@ -426,56 +444,67 @@ subroutine validate_inflate_options(inf_flavor, inf_damping, inf_initial_from_re
-if (inf_flavor(1) > 1) do_prior_inflate     = .true.
-if (inf_flavor(2) > 1) do_posterior_inflate = .true.
+if (inf_flavor(PRIOR_INF)     /= NO_INFLATION .and. &
+    inf_flavor(PRIOR_INF)     /= OBS_INFLATION)  do_prior_inflate     = .true.
+if (inf_flavor(POSTERIOR_INF) /= NO_INFLATION .and. &
+    inf_flavor(POSTERIOR_INF) /= OBS_INFLATION)  do_posterior_inflate = .true.
+
 if (do_prior_inflate .or. do_posterior_inflate) output_inflation = .true.
 
 ! Observation space inflation not currently supported
-if(inf_flavor(1) == 1 .or. inf_flavor(2) == 1) &
+if(inf_flavor(PRIOR_INF) == OBS_INFLATION .or. &
+   inf_flavor(POSTERIOR_INF) == OBS_INFLATION) &
    call error_handler(E_ERR, 'validate_inflate_options', &
    'observation space inflation (type 1) not currently supported', source, &
    text2='contact DART developers if you are interested in using it.')
 
 ! Relaxation-to-prior-spread (RTPS) is only an option for posterior inflation
-if(inf_flavor(1) == 4) &
+if(inf_flavor(PRIOR_INF) == RELAXATION_TO_PRIOR_SPREAD) &
    call error_handler(E_ERR, 'validate_inflate_options', &
    'RTPS inflation (type 4) only supported for Posterior inflation', source)
 
 ! Cannot select posterior options if not computing posterior
-if(.not. compute_posterior .and. inf_flavor(2) > 0) then
+if(.not. compute_posterior .and. inf_flavor(POSTERIOR_INF) /= NO_INFLATION) then
    write(string1, *) 'cannot enable posterior inflation if not computing posterior values'
    call error_handler(E_ERR,'validate_inflate_options', string1, source, &
-                             text2='"compute_posterior" is false; posterior inflation flavor must be 0')
+           text2='"compute_posterior" is false; posterior inflation flavor must be 0')
 endif
 
 ! RTPS needs a single parameter from namelist: inf_initial(2).  
-! Do not read in any files.  Also, no damping.  but warn the user if they try to set different
-! values in the namelist.
-if (inf_flavor(2) == 4) then
-   if (inf_initial_from_restart(2) .or. inf_sd_initial_from_restart(2)) &
-      call error_handler(E_MSG, 'validate_inflate_options:', &
-         'RTPS inflation (type 4) overrides posterior inflation restart file with value in namelist', &
-         text2='posterior inflation standard deviation value not used in RTPS')
-   inf_initial_from_restart(2) = .false.    ! Get parameter from namelist inf_initial(2), not from file
-   inf_sd_initial_from_restart(2) = .false. ! inf_sd not used in this algorithm
+! Do not read in any files.  Also, no damping.  
+! Warn the user if they try to set different values in the namelist.
+if (inf_flavor(POSTERIOR_INF) == RELAXATION_TO_PRIOR_SPREAD) then
+
+   if ( inf_sd_initial_from_restart(POSTERIOR_INF)) &
+       call error_handler(E_MSG, 'validate_inflate_options:', &
+         'RTPS inflation (type 4) does not use "inf_sd_inifial_from_restart" values.', &
+         text2='values of "sd" in output inflation variables are not meaningful.')
```

Comment:
I disagree ... the message stated that "... overrides the posterior inflation restart **file** with value in namelist".
the RTPS PR changed that ... the posterior inflation file now has the actual inflation values - no matter what the namelist values were.

Reply:
Since RTPS does not use the inf_sd values at all ... it does parrot the sd namelist values into the output inflation file.
No matter what the 'mean' inflation values are in the namelist - or the input inflation files - they are completely disjoint from the inflation values calculated and used by RTPS ...

Reply:
The message is a warning about reading *from* inflation files,   if you have 

````
&filter_nml
inf_initial_from_restart = .true.
````

RTPS will not read the values from the file:

````
   ! Get parameter from namelist inf_initial(2), not from file
   ! inf_sd not used in this algorithm
      inf_initial_from_restart(POSTERIOR_INF) = .false.
     inf_sd_initial_from_restart(POSTERIOR_INF) = .false.
````

I think changing error messages for RTPS is outside the scope of this pull request,  I recommend 
switching out '2' for POSTERIOR_INF and leaving the rest of the message unchanged.

Reply:
Helen, perhaps as you say this might not be in the scope of this PR. However, Tim is right. The message is wrong. After the correction that I made, this error message should be updated as Tim suggests. In fact, if I was more careful I should've done it in my PR originally.

Reply:
Moha, can you create an issue for what the problem with the message is. At the moment it is not clear to me how your correction https://github.com/NCAR/DART/pull/165 relates to warning users about inflation files not being read in.   I thought it was a key point for users to be aware that
"RTPS inflation does not use any files (i.e., inflation has no-memory between DA cycles).:

Let's leave any RTPS changes out of this pull request and fix and RTPS message problems separately.

Reply:
i think these messages are about the *INPUT* files ( "from_restart" ).  didn't moha's changes only apply to the files it's writing out?

Reply:
p.s. didn't we remove the namelist options to control writing out inflation restart files?  you can control if the values are read from a file or from the namelist, but if inflation is on you get restart files whether you want them or not.